src/app/modules/courses/services/courses.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Course } from '@modules/courses/types/course';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { coursesMock } from '@app/testing/courses.mock';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CoursesService {
+
+  private _courses = new BehaviorSubject<Course[]>([...coursesMock]);
+
+  constructor() { }
+
+  getCourses(): Observable<Course[]> {
+    return this._courses.asObservable();
+  }
+
+  createCourse(course: Omit<Course, 'id'>): void {
+    const courses = this._courses.getValue();
+    courses.sort(item => item.id);
+    const highestIndex = Math.max(...courses.map(item => item.id));
+    course['id'] = highestIndex + 1;
```

Comment:
For it looks a bit hard,  because you can just generate random number here and use it as ID. But anyway this part will be removed in future MRs, so I don't mind about such solutionn

Reply:
ok